src/CircuitBreaker.ts

```diff
@@ -62,12 +68,15 @@ export class CircuitBreaker extends EventEmitter {
     }
 
     async _evaluate() {
-        this._evaluatingPromiseHook = Promise.allSettled(this._probes.map(p => p.check()));
+        this._evaluatingPromiseHook = Promise.allSettled(this._probes.map(p => p.check().catch(() => {
+            this._cbError = true;
+        })));
         await this._evaluatingPromiseHook;
         this._evaluatingPromiseHook = null;
-
+        if (!this._cbError) {
+            this._cbError = false;
+        }
```

Comment:
how can this happen?
you initialize in the constructor, so cannot be undefined...